### DISCUSSION TITLE
navigator.plugins suggestion

### DISCUSSION THREAD

**Discussion by r-a-y** [Eloston/ungoogled-chromium#1363] [Ideas]
The `pdf-plugin-name` flag is used to set the PDF flag depending on the browser we want to set it to. When set to `chrome`, this changes the PDF plugins from `Chromium` to `Chrome` for `navigator.plugins` so UC looks more like Chrome.

Should we also have another flag to add the Native Client (`internal-nacl-plugin`) from Google Chrome to `navigator.plugins`?

![native-client](https://user-images.githubusercontent.com/505921/106211729-1a22c200-61c1-11eb-9f25-7646ce3a8b63.png)

This would make Ungoogled Chromium look more like Chrome and would prevent our fingerprint from sticking out too much. We don't necessarily need to implement Native Client support, but maybe a dummy would be enough.

Thoughts?

**Comment by Eloston** [comment]
Is this still needed on version 88? They've changed the plugin code, so I'm not sure what the status is for NaCl.

**Reply by r-a-y** [reply]
Yes it is. The screenshot above was taken from Google Chrome 88.

**Reply by r-a-y** [reply]
This might be a good base to use for spoofing `navigator.plugins` and `navigator.mimeTypes`:

https://github.com/berstend/puppeteer-extra/tree/master/packages/puppeteer-extra-plugin-stealth/evasions/navigator.plugins
 (via https://stackoverflow.com/a/54373605)

**Reply by r-a-y** [reply]
Actually, looks like `navigator.plugins` is going to return nothing in v90:
https://bugs.chromium.org/p/chromium/issues/detail?id=1164635

Can probably wait until v90 and we won't have to do a thing!

**Reply by r-a-y** [reply]
Just to update, sadly the Chromium devs decided to revert a change to implement the blank array for `navigator.plugins` in v90 due to PDF support not working in some instances: https://bugs.chromium.org/p/chromium/issues/detail?id=1164635#c40

More info here: https://github.com/whatwg/html/issues/6003#issuecomment-805389170

From the WhatWG issue, it sounds like we might not need to do anything as they are looking at hardcoding the PDF plugin only.
